### ISSUE TITLE
cylc stop doesn't work when reg ends with slash

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#4010]
**TL;DR**

When you do `cylc run`, if you have a slash at the end of the workflow name, e.g. `cylc run foo/`, then `cylc stop foo` won't work, only `cylc stop foo/` will work. And vice versa.

**Release version(s) and/or repository branch(es) affected?**
Cylc 8.0a3.dev master

**Steps to reproduce the bug**
```console
$ cylc run foo --hold
$ cylc stop foo/  # appears to work (no output as is the case of a success)
$ cylc get-suite-contact foo
CYLC_API=5
CYLC_SUITE_PROCESS=1234 /path/to/python3 /path/to/cylc run foo/ --hold
# etc
```

**Expected behavior**
Workflow stops and contact file removed.

**Additional context**
Often when I'm in the cylc-run dir, I use <kbd>tab</kbd> autocompletion to run workflows, stop them etc. But because the workflows are directories, that adds a `/` to the end of the reg (workflow name).

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

**Comment by oliver-sanders** [comment]
I'm unable to replicate this, did you give `foo` sufficient time to recieve the stop signal and shut down?

**Comment by oliver-sanders** [comment]
Confirmed by mixing use of trailing slash with `cylc run` and `cylc stop`:

```console
$ cylc run --hold g1/

$ cylc run --hold g2

$ cylc stop g1; echo $?
0

$ cylc stop g2; echo $?
0

$ cylc cat-log g1
# ...
2020-12-22T13:48:07Z INFO - Final point: None
2020-12-22T13:48:07Z INFO - (foo.20191209T1200Z) spawned foo.20191210T0000Z flow(w)
2020-12-22T13:48:07Z INFO - (foot.20191209T1200Z) spawned foot.20191210T0000Z flow(w)
2020-12-22T13:48:07Z INFO - (fool.20191209T1200Z) spawned fool.20191210T0000Z flow(w)
2020-12-22T13:48:17Z INFO - [client-command] graphql osanders@vld601:cylc

$ cylc cat-log g2
# ...
2020-12-22T13:48:11Z INFO - Final point: None
2020-12-22T13:48:11Z INFO - (foo.20191209T1200Z) spawned foo.20191210T0000Z flow(y)
2020-12-22T13:48:11Z INFO - (foot.20191209T1200Z) spawned foot.20191210T0000Z flow(y)
2020-12-22T13:48:11Z INFO - (fool.20191209T1200Z) spawned fool.20191210T0000Z flow(y)
2020-12-22T13:48:23Z INFO - [client-command] graphql osanders@vld601:cylc
2020-12-22T13:48:23Z INFO - Command succeeded: stop(mode=StopMode.REQUEST_CLEAN)
2020-12-22T13:48:23Z INFO - Processing 1 queued command(s)
	+       stop(mode=StopMode.REQUEST_CLEAN)
2020-12-22T13:48:23Z INFO - Suite shutting down - REQUEST(CLEAN)
2020-12-22T13:48:23Z WARNING - Some partially satisfied prerequisites left over:
	foo.20191210T0000Z is waiting on:
	* foo_start.20191209T0900Z succeeded
	* cycle_end.20191209T1200Z succeeded
	foot.20191210T0000Z is waiting on:
	* foo_start.20191209T0900Z succeeded
	* cycle_end.20191209T1200Z succeeded
	fool.20191210T0000Z is waiting on:
	* foo_start.20191209T0900Z succeeded
	* cycle_end.20191209T1200Z succeeded
2020-12-22T13:48:24Z INFO - DONE
```

It looks like the `.service` dirs are configured differently:

```console
$ tree g* -a
g1
|-- .service
|   |-- client.key_secret
|   |-- client_public_keys
|   |   `-- client_localhost.key
|   |-- contact
|   |-- db
|   |-- etc
|   |   `-- job.sh
|   |-- server.key
|   `-- server.key_secret
|-- flow.cylc -> .../g1/suite.rc
|-- flow.cylc.processed
|-- log
|   |-- db
|   |-- flow-config
|   |   `-- 20201222T134807Z-run.cylc
|   |-- job
|   `-- suite
|       |-- file-installation-log
|       |-- log -> log.20201222T134807Z
|       `-- log.20201222T134807Z
|-- share
|-- suite.rc
`-- work
g2
|-- .service
|   |-- db
|   `-- etc
|       `-- job.sh
|-- flow.cylc -> .../g2/suite.rc
|-- flow.cylc.processed
|-- log
|   |-- db
|   |-- flow-config
|   |   `-- 20201222T134811Z-run.cylc
|   |-- job
|   `-- suite
|       |-- file-installation-log
|       |-- log -> log.20201222T134811Z
|       `-- log.20201222T134811Z
|-- share
|-- suite.rc
`-- work
```

**Comment by oliver-sanders** [comment]
Have checked, this does not effect Cylc7.

**Comment by oliver-sanders** [comment]
In the not too distant future the [universal id](https://github.com/cylc/cylc-flow/pull/3931) will strip away trailing slashes from all input IDs, however, it would still be good to know this can't resurface.
